src/util/CompositeAsyncHandler.ts

```diff
@@ -86,8 +87,35 @@ export class CompositeAsyncHandler<TIn, TOut> implements AsyncHandler<TIn, TOut>
       }
     }
 
+    let differentCodes = false;
+    const last = { statusCode: 0, name: '' };
+
+    for (const error of errors) {
+      if (error instanceof HttpError) {
+        if (last.statusCode !== error.statusCode) {
+          if (last.statusCode > 0) {
+            // Don't break yet since there might still be a 500 error in there
+            differentCodes = true;
+          } else {
+            last.statusCode = error.statusCode;
+            last.name = error.name;
+          }
+        }
+      } else {
+        last.statusCode = 500;
+        break;
+      }
+    }
+
```

Comment:
I find the algorithm a bit weird (but it seems to work).

Reply:
I'd just collect status codes in an array then with a `map`, and do these operations on that array.

It's not as much nits about the code as it it about the fact that we have to make it this complicated; it might be a sign of something else being wrong (like the "normal Error = 500" bit being implicitly coded here).

Reply:
Regarding the code: could also write it like something as follows should that be better
```typescript
if (errors.some(error => error.statusCode === 500))
  throw 500 error;
if (all errors have same status code)
  throw that specific error;
throw 400 error;
```
I just wrote it like I did to only have to loop only once through the array but in hindsight the above would probably be clearer.


The normal error = 500 thing is something that we need to have anyway, since we actually never want 500 errors, and they are usually an indication that something somewhere threw an error that we didn't catch (because it's not supposed to throw an error in normal circumstances, e.g. `modifyResource` being called on the file store). I could explicitly create an HttpError with 500 here if that would be clearer.


I also had #6 in mind when writing this, but this doesn't actually change anything about this issue. Should we change how we handle errors this can easily be changed here as well. But I don't want to let #23 hang because of that since it's seems important to have correct status codes and it only requires a small change to do so.